lib/coinbase/staking_reward.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+require 'date'
+
+module Coinbase
+  # A representation of a staking reward earned on a network for a given asset.
+  class StakingReward
+    # Returns a list of StakingRewards for the provided network, asset, and addresses.
+    # @param network_id [Symbol] The network ID
+    # @param asset_id [Symbol] The asset ID
+    # @param address_ids [Array<String>] The address IDs
+    # @param start_time [Time] The start time. Defaults to one month ago.
+    # @param end_time [Time] The end time. Defaults to the current time.
+    # @param format [Symbol] The format to return the rewards in. (:usd, :native) Defaults to :usd.
+    # @return [Enumerable<Coinbase::StakingReward>] The staking rewards
+    def self.list(network_id, asset_id, address_ids, start_time: DateTime.now.prev_month(1), end_time: DateTime.now,
+                  format: :usd)
+      Enumerator.new do |yielder|
+        asset = Coinbase.call_api do
+          Asset.fetch(network_id, asset_id)
+        end
+
+        page = nil
+
+        loop do
+          resp = Coinbase.call_api do
+            req = {
+              network_id: Coinbase.normalize_network(network_id),
+              asset_id: asset_id,
+              address_ids: address_ids,
+              start_time: start_time.iso8601,
+              end_time: end_time.iso8601,
+              format: format,
+              next_page: page
+            }
+            stake_api.fetch_staking_rewards(req)
+          end
+
+          resp.data.each do |staking_reward|
+            yielder << new(staking_reward, asset, format)
+          end
+
+          break unless resp.has_more
```

Comment:
We should set the `page` explicitly, or use the Pagination helper:

https://github.com/coinbase/coinbase-sdk-ruby/blob/6fb94d565dbe45f2eb9eed3157f675bd9de28aaf/lib/coinbase/pagination.rb#L19-L21